plugins/aws/resoto_plugin_aws/resource/apigateway.py

```diff
@@ -0,0 +1,447 @@
+    stage_access_log_settings: Dict[str, str] = field(default=None)
+    stage_canary_settings: Optional[AwsApiGatewayCanarySetting] = field(default=None)
+    stage_tracing_enabled: bool = field(default=None)
+    stage_web_acl_arn: Optional[str] = field(default=None)
+    api_link: str = field(default=None)
+
+    # TODO add edge to Web Acl when applicable (via stage_web_acl_arn)
+
+    def delete_resource(self, client: AwsClient) -> bool:
+        client.call(
+            service="apigateway",
+            action="delete-stage",
+            result_name=None,
+            restApiId=self.api_link,
+            stageName=self.name,
+        )
+        return True
+
+
+@define(eq=False, slots=False)
+class AwsApiGatewayDeployment(AwsResource):
+    kind: ClassVar[str] = "aws_api_gateway_deployment"
+    reference_kinds: ClassVar[ModelReference] = {"successors": {"default": ["aws_api_gateway_stage"]}}
+
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "ctime": S("createdDate"),
+        "description": S("description"),
+        "deployment_api_summary": S("apiSummary"),
+    }
+    description: Optional[str] = field(default=None)
+    deployment_api_summary: Dict[str, Dict[str, Dict[str, Union[str, bool]]]] = field(default=None)
+    api_link: str = field(default=None)
+
+    def delete_resource(self, client: AwsClient) -> bool:
+        client.call(
+            service="apigateway",
+            action="delete-deployment",
+            result_name=None,
+            restApiId=self.api_link,
+            deploymentId=self.id,
+        )
+        return True
+
+
+@define(eq=False, slots=False)
+class AwsApiGatewayEndpointConfiguration:
+    kind: ClassVar[str] = "aws_api_gateway_endpoint_configuration"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "types": S("types", default=[]),
+        "vpc_endpoint_ids": S("vpcEndpointIds", default=[]),
+    }
+    types: List[str] = field(factory=list)
+    vpc_endpoint_ids: List[str] = field(factory=list)
+
+
+@define(eq=False, slots=False)
+class AwsApiGatewayRestApi(ApiGatewayTaggable, AwsResource):
+    kind: ClassVar[str] = "aws_api_gateway_rest_api"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("apigateway", "get-rest-apis", "items")
+    reference_kinds: ClassVar[ModelReference] = {
+        "successors": {
+            "default": [
+                "aws_vpc_endpoint",
+                "aws_api_gateway_deployment",
+                "aws_api_gateway_authorizer",
+                "aws_api_gateway_resource",
+            ],
+            "delete": ["aws_vpc_endpoint"],
+        }
+    }
+
+    @classmethod
+    def called_apis(cls) -> List[AwsApiSpec]:
+        return [
+            cls.api_spec,
+            AwsApiSpec("apigateway", "get-deployments"),
+            AwsApiSpec("apigateway", "get-stages"),
+            AwsApiSpec("apigateway", "get-authorizers"),
+            AwsApiSpec("apigateway", "get-resources"),
+        ]
```

Comment:
let's try to have a section for properties and a section for methods.
mixing them up, makes it hard to read.